lib/ansible/galaxy/collection.py

```diff
@@ -356,6 +378,45 @@ def from_name(collection, apis, requirement, force, parent=None):
         return req
 
 
+def get_available_api_versions(galaxy_api):
```

Comment:
As I mentioned above, this seems like it should be part of the `GalaxyAPI` class and not here. You could make it so it only collections the versions when someone tries to access the `available_api_version` property lazily to avoid running it multiple times on the same server.